app/models/shopify/import/base.rb

```diff
@@ -0,0 +1,69 @@
+module Shopify
+  module Import
+    class Base
+      def initialize(credentials: {}, client: nil)
+        @client = client(credentials) unless client
+      end
+
+      def count(**opts)
+        api_class.get(:count, opts)
+      end
+
+      def find(id)
+        api_class.find(id)
+      end
+
+      def find_and_import(id)
+        resource = find(id)
+        create_data_feed_and_spree_object(resource)
+      end
+
+      def find_all(**opts)
+        results = []
+        find_in_batches(**opts) do |batch|
+          return results if batch.nil?
+          batch.each { |resource| results << resource }
+        end
+        results
+      end
+
+      def find_all_first_and_import(**opts)
+        find_all(**opts).each do |resource|
+          create_data_feed_and_spree_object(resource)
+        end
+      end
+
+      private
+
+      def find_in_batches(**opts)
+        opts = { limit: 250, page: 1 }.merge(opts)
+        begin
+          batch = api_class.find(:all, params: opts)
+          yield batch
+          opts[:page] += 1
+        end while batch.try(:any?)
+      end
+
+      def client(credentials)
+        Shopify::Import::Client.instance.get_connection(credentials)
+      end
+
+      def create_data_feed_and_spree_object(resource)
```

Comment:
This class shouldn't be doing this. Its main functionality is to fetch stuff from Shopify API. This makes it:

* fetch stuff from Shopify
* call the parser of the data
* store the result